Improve generate-ref-docs prerequisites (Go version, drop GOPATH/go get) (#56403)
kubernetes/website: content/en/docs/contribute/generate-ref-docs/prerequisites-ref-docs.md

Reviewer: Since this is the shared prerequisites, could we add the Windows path here? 

The toolchain is Unix-only (make + the bash hack/*.sh scripts), so Windows contributors need WSL. A one-liner on the "Linux or macOS" line , "on Windows, use WSL" means every generate-ref-docs page inherits it.

Author: Added, and linked the WSL installation doc

### Requirements:

- You need a machine that is running Linux or macOS. On Windows, use
  [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install),
  since the build tooling relies on `make` and Bash scripts.

- You need to have these tools installed:

  - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
  - [Go](https://go.dev/dl/), any recent release (Go downloads the exact toolchain a generator needs automatically)
  - [make](https://www.gnu.org/software/make/)
  - [gcc compiler/linker](https://gcc.gnu.org/)

>   - [Docker](https://docs.docker.com/engine/installation/) (only required by the deprecated `gen-kubectldocs` generator, or if you want to preview your changes locally with `make container-serve`)

Reviewer: ```suggestion
  - [Docker](https://docs.docker.com/engine/installation/) (required only for the local website preview with `make container-serve`)
```

- You need to have these tools installed:

  - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)

>   - [Golang](https://go.dev/dl/), matching the version required by the `go.mod` file of the `reference-docs` module you're building (`gen-apidocs`, `gen-compdocs`, or `genref`)

Reviewer: Nice fix over the hardcoded floor. One tweak: a contributor shouldn't have to match a version per generator, with Go's default GOTOOLCHAIN=auto, a single recent install works everywhere;

```suggestion
- [Go](https://go.dev/dl/), any recent release (Go downloads the exact toolchain a generator needs automatically)
```

Author: thanks, applied as suggested

  - [make](https://www.gnu.org/software/make/)
  - [gcc compiler/linker](https://gcc.gnu.org/)
  - [Docker](https://docs.docker.com/engine/installation/) (only required for the deprecated `gen-kubectldocs` generator)

>   - [Python](https://www.python.org/downloads/) v3.7.x+, [Pip](https://pypi.org/project/pip/), and [PyYAML](https://pyyaml.org/) v5.1.2 (only required to run the deprecated `update-imported-docs.py` script)

Reviewer: Docker is only for the deprecated gen-kubectldocs (the current kubectl reference uses gen-compdocs, pure Go). Same rationale as dropping Python below. I'd drop these two line, so the list only covers the supported generators (Git, Go, make, gcc).

Reviewer 2: You might well want Docker to test that the site builds OK, though.

Reviewer: Yeah, that's true

Author: dropped the Python/Pip/PyYAML line entirely, but kept Docker since it's still what make container-serve uses for local preview

  The `reference-docs` generators are Go modules, so you don't need to set a `GOPATH` or use `go get`.

> - Your `PATH` environment variable must include the required build tools, such as the `Go` binary.

Reviewer: I would also drop these entirely. No need to list dependencies for a deprecated script; scoping the list to the current flow keeps it forward-looking.

Author: done
